### PR TITLE
Retrieve device based on deviceNumber

### LINKED ISSUE #187: Network* objects are relying on array index instead of deviceNumber
The process of building a NetworkInterface should rely on the deviceId to identify which adapter will be used. But it is using deviceId as index on the collection of adapters (as shown [here](https://github.com/HewlettPackard/oneview-redfish-toolkit/blob/master/oneview_redfish_toolkit/api/network_interface.py#L42)) instead of retrieving the device based on deviceNumber following the indexing rule [at the collection](https://github.com/HewlettPackard/oneview-redfish-toolkit/blob/master/oneview_redfish_toolkit/api/network_interface_collection.py#L51). Also happening on: [NetworkAdapters](https://github.com/HewlettPackard/oneview-redfish-toolkit/blob/master/oneview_redfish_toolkit/api/network_adapter.py#L45); [NetworkPortCollection](https://github.com/HewlettPackard/oneview-redfish-toolkit/blob/master/oneview_redfish_toolkit/api/network_port_collection.py#L69); [NetworkPort](https://github.com/HewlettPackard/oneview-redfish-toolkit/blob/master/oneview_redfish_toolkit/api/network_port.py#L45); [NetworkDeviceFunctionCollection](https://github.com/HewlettPackard/oneview-redfish-toolkit/blob/master/oneview_redfish_toolkit/api/network_device_function_collection.py#L43); and [NetworkDeviceFunction](https://github.com/HewlettPackard/oneview-redfish-toolkit/blob/master/oneview_redfish_toolkit/api/network_device_function.py#L46). A common fix may be applied to all these cases.

### PR DESCRIPTION
**Pull request by llvieira**
Closes #187 and #171.

- Tests created.
- Fixed all retrieval of devices in network objects.

### CODE REVIEW

File: oneview_redfish_toolkit/api/network_device_function.py [hunk 1]
```diff
@@ -43,21 +43,30 @@ def __init__(self, device_id, device_function_id, server_hardware):
...
+
+            port = self.get_resource_by_id(
+                physical_ports, "portNumber",
+                port_number, "Network Device Function")
+
+            virtual_port = self.get_resource_by_id(
+                port["virtualPorts"], "portNumber",
+                virtual_port_number, "Network Device Function")
+
+            virtual_port_function_validation = virtual_port["portFunction"]
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/api/network_device_function.py]: not necessary, line 60 validates this

File: oneview_redfish_toolkit/api/network_device_function.py [hunk 2]
```diff
@@ -43,21 +43,30 @@ def __init__(self, device_id, device_function_id, server_hardware):
...
         except Exception:
             raise OneViewRedfishResourceNotFoundError(
                 device_function_id, "NetworkDeviceFunction")
 
-        if virtual_port_function_validation != virtual_port_funcion:
+        if virtual_port_function_validation != virtual_port_function:
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/api/network_device_function.py]: see comment above

File: oneview_redfish_toolkit/api/network_port.py [hunk 3]
```diff
@@ -42,30 +40,13 @@ def __init__(self, device_id, port_id, server_hardware):
...
-            if port_index == -1:
-                raise OneViewRedfishResourceNotFoundError(
-                    port_id, "NetworkPort")
-        except Exception:
-            raise OneViewRedfishResourceNotFoundError(
-                port_id, "NetworkPort")
 
-        port = server_hardware["portMap"]["deviceSlots"][index][
-            "physicalPorts"][port_index]
+        physical_ports = self.get_resource_by_id(
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/api/network_port.py]: you killed the tests made at lines 54-59. you need to validate port type.

File: oneview_redfish_toolkit/api/redfish_json_validator.py [hunk 4]
```diff
@@ -96,3 +96,36 @@ def serialize(self):
...
+        try:
+            resource_id = int(resource_id)
+        except ValueError:
+            raise OneViewRedfishError("Invalid {} ID".format(resource_name))
+
+        for resource in resource_list:
+            if resource[resource_number_key] == resource_id:
+                return resource
+
+        return None
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/api/redfish_json_validator.py]: this is not acceptable. When this function is called it MUST!!!! return a valid resource or should raise an exception.

File: oneview_redfish_toolkit/api/network_device_function.py (hunk 1 above)
Comment (llvieira) [reply oneview_redfish_toolkit/api/network_device_function.py]: Fixed.

File: oneview_redfish_toolkit/api/network_device_function.py (hunk 2 above)
Comment (llvieira) [reply oneview_redfish_toolkit/api/network_device_function.py]: Fixed.

File: oneview_redfish_toolkit/api/redfish_json_validator.py (hunk 4 above)
Comment (llvieira) [reply oneview_redfish_toolkit/api/redfish_json_validator.py]: Fixed.

File: oneview_redfish_toolkit/api/network_port.py (hunk 3 above)
Comment (llvieira) [reply oneview_redfish_toolkit/api/network_port.py]: Fixed.

File: oneview_redfish_toolkit/api/redfish_json_validator.py [hunk 5]
```diff
@@ -96,3 +98,37 @@ def serialize(self):
             default=lambda o: o.__dict__,
             sort_keys=False,
             indent=indent)
+
+    def get_resource_by_id(self, resource_list, resource_number_key,
+                           resource_id, resource_name):
```
[the remark this answers is not in the document]
Comment (llvieira) [reply oneview_redfish_toolkit/api/redfish_json_validator.py]: Fixed!

**Pull request merged**